clientconn.go

```diff
@@ -248,16 +248,26 @@ func Dial(target string, opts ...DialOption) (*ClientConn, error) {
 			return nil, errNoAddr
 		}
 	}
+
+	errC := make(chan error, len(addrs))
```

Comment:
because if the first message over `errC` is non-nil, `Dial` will fail; the intended behavior of this patch is if any connection succeeds within the timeout, dial will succeed as well